go/sync2/semaphore.go

```diff
@@ -66,3 +66,8 @@ func (sem *Semaphore) TryAcquire() bool {
 func (sem *Semaphore) Release() {
 	sem.slots <- struct{}{}
 }
+
+// Size returns the current number of available slots.
+func (sem *Semaphore) Size() int {
+	return len(sem.slots)
```

Comment:
I think this racy, unless the language guarantees it. I couldn't find any text about this.